combine/combine_test.go

```diff
@@ -0,0 +1,253 @@
+		})
+
+		Convey("You can concatenate the inputs to compressed output", func() {
+			compressor, closer, err := Compress(output)
+			So(err, ShouldBeNil)
+
+			err = Concatenate(inputs, compressor)
+			So(err, ShouldBeNil)
+
+			closer()
+
+			b, err := os.ReadFile(outputPath)
+			So(err, ShouldBeNil)
+			So(string(b), ShouldNotEqual, "line from path1\nline from path2\n")
+
+			actualFileContents, err := ReadCompressedFile(outputPath)
+			So(err, ShouldBeNil)
+			So(actualFileContents, ShouldEqual, "line from path1\nline from path2\n")
+		})
+
+		Convey("You can concatenate the inputs to compressed output in a single method call", func() {
+			err := ConcatenateAndCompress(inputs, output)
+			So(err, ShouldBeNil)
+
+			b, err := os.ReadFile(outputPath)
+			So(err, ShouldBeNil)
+			So(string(b), ShouldNotEqual, "line from path1\nline from path2\n")
+
+			actualFileContents, err := ReadCompressedFile(outputPath)
+			So(err, ShouldBeNil)
+			So(actualFileContents, ShouldEqual, "line from path1\nline from path2\n")
+		})
+
+		Convey("You can merge the inputs to the output", func() {
+			merger := myMerger
+
+			err := Merge(inputs, output, merger)
+			So(err, ShouldBeNil)
+
+			b, err := os.ReadFile(outputPath)
+			So(err, ShouldBeNil)
+			So(string(b), ShouldEqual, "line from path1\nline from path2\n")
+		})
+
+		Convey("You can merge and compress the inputs to the output", func() {
+			merger := myMerger
+
+			compressor, closer, err := Compress(output)
+			So(err, ShouldBeNil)
+
+			err = Merge(inputs, compressor, merger)
+			So(err, ShouldBeNil)
+
+			closer()
+
+			b, err := os.ReadFile(outputPath)
+			So(err, ShouldBeNil)
+			So(string(b), ShouldNotEqual, "line from path1\nline from path2\n")
+
+			actualFileContents, err := ReadCompressedFile(outputPath)
+			So(err, ShouldBeNil)
+			So(actualFileContents, ShouldEqual, "line from path1\nline from path2\n")
+		})
+
+		Convey("You can merge and compress the inputs to the output in a single call", func() {
+			merger := myMerger
+
+			err := MergeAndCompress(inputs, output, merger)
+			So(err, ShouldBeNil)
+
+			b, err := os.ReadFile(outputPath)
+			So(err, ShouldBeNil)
+			So(string(b), ShouldNotEqual, "line from path1\nline from path2\n")
+
+			actualFileContents, err := ReadCompressedFile(outputPath)
+			So(err, ShouldBeNil)
+			So(actualFileContents, ShouldEqual, "line from path1\nline from path2\n")
+		})
+
+		Convey("MergeSummaryLines merges the file contents to output", func() {
+			inputFiles := make([]string, len(inputs))
+			for i, file := range inputs {
+				inputFiles[i] = file.Name()
+			}
+
+			sortMergeOutput, cleanup, err := mergeSortedFiles(inputFiles)
+			So(err, ShouldBeNil)
+
+			err = MergeSummaryLines(sortMergeOutput, 3, 2, myLineMerger, output)
+			So(err, ShouldBeNil)
+
+			err = cleanup()
+			So(err, ShouldBeNil)
+
+			b, err := os.ReadFile(outputPath)
+			So(err, ShouldBeNil)
+
+			So(string(b), ShouldEqual, "line from path1\nline from path2\n")
```

Comment:
This isn't a good test of MergeSummaryLines. It doesn't test that it does anything with 3, 2 args.